src/toil_scripts/gatk_processing/gatk_preprocessing.py

```diff
@@ -521,91 +330,111 @@ def indel_realignment(job, shared_ids, input_args):
-    job.addChildJobFn(base_recalibration, shared_ids, input_args, cores = input_args.cpu_count)
+    bam_id = job.fileStore.writeGlobalFile(outpath_bam)
+    bai_id = job.fileStore.writeGlobalFile(outpath_bai)
+    return bam_id, bai_id
 
 
-def base_recalibration(job, shared_ids, input_args):
+def gatk_base_recalibrator(job, bamFileStoreID, baiFileStoreID, config, xmx=8):
     """
     Creates recal table to perform Base Quality Score Recalibration
 
-    job_vars: tuple     Contains the input_args and ids dictionaries
-    sample: str         Either "normal" or "tumor" to track which one is which
+    :param job:
+    :param bamFileStoreID:
+    :param baiFileStoreID:
+    :param config:
+    :param xmx:
+    :return:
     """
-    # Unpack convenience variables for job
+    job.fileStore.logToMaster('Running GATK BaseRecalibrator: {}'.format(config['uuid']))
     work_dir = job.fileStore.getLocalTempDir()
+    cores = multiprocessing.cpu_count()
     # Retrieve input file paths
-    return_input_paths(job, work_dir, shared_ids, 'ref.fa', 'sample.indel.bam',
-                       'dbsnp.vcf', 'ref.fa.fai',
-                       'ref.dict', 'sample.indel.bam.bai')
+    references = ['genome.fa','genome.fa.fai', 'genome.dict', 'dbsnp.vcf']
+    inputs = {'sample.indel.bam': bamFileStoreID, 'sample.indel.bam.bai': baiFileStoreID}
+    inputs.update({key: config[key] for key in references})
+    get_files_from_filestore(job, work_dir, inputs)
     # Output file path
     output = os.path.join(work_dir, 'sample.recal.table')
+
     # Call: GATK -- IndelRealigner
     parameters = ['-U', 'ALLOW_SEQ_DICT_INCOMPATIBILITY', # RISKY! (?) See #189
                   '-T', 'BaseRecalibrator',
-                  '-nct', str(input_args.cpu_count),
-                  '-R', 'ref.fa',
+                  '-nct', str(cores),
+                  '-R', 'genome.fa',
                   '-I', 'sample.indel.bam',
                   '-knownSites', 'dbsnp.vcf',
                   '-o', 'sample.recal.table']
     docker_call(tool='quay.io/ucsc_cgl/gatk:3.5--dba6dae49156168a909c43330350c6161dc7ecc2',
                 work_dir=work_dir, parameters=parameters,
-                inputs=['ref.fa', 'sample.indel.bam', 'dbsnp.vcf', 'ref.fa.fai',
-                        'ref.dict', 'sample.indel.bam.bai'],
+                inputs=inputs.keys(),
                 outputs={'sample.recal.table': None},
-                env={'JAVA_OPTS':'-Xmx%sg' % input_args.memory})
+                env={'JAVA_OPTS':'-Xmx%sg' % xmx},
+                mock=config['mock'])
     # Write to fileStore
-    shared_ids['sample.recal.table'] = job.fileStore.writeGlobalFile(output)
-    job.addChildJobFn(print_reads, shared_ids, input_args, cores = input_args.cpu_count)
+    return job.fileStore.writeGlobalFile(output)
 
 
-def print_reads(job, shared_ids, input_args):
+def gatk_print_reads(job, bamFileStoreID, baiFileStoreID, recalFileStoreID, config, xmx=8):
```

Comment:
This exists in `toil_scripts.tools.preprocessing` already; remove/consolidate differences.